xmppserver/src/main/java/org/jivesoftware/openfire/handler/IQVersionHandler.java

```diff
@@ -76,6 +83,40 @@ public IQ handleIQ(IQ packet) throws PacketException {
             IQ result = IQ.createResultIQ(packet);
             result.setError(PacketError.Condition.not_acceptable);
             return result;
+        } else if (IQ.Type.result == packet.getType()) {
+            /* handle results coming through BOSH Connections,
+             * other results are processed in org.jivesoftware.openfire.net.SocketRead.java - getIQ()
+             */
+            try {
+                LocalSession localSession = (LocalSession) XMPPServer.getInstance().getSessionManager().getSession(packet.getFrom());
+
+                Element query = packet.getChildElement();
+                List<Element> elements = query.elements();
+                if (elements.size() > 0) {
+                    for (Element element : elements) {
+                        if (element.getName() != null && element.getStringValue() != null) {
+                            if (localSession!=null)
+                            {
+                                localSession.setSoftwareVersionData(element.getName(), element.getStringValue());
+                            }
+                            else
+                            {
+                                /*
+                                  The result comes from a server 2 server connection, so we write the information
+                                  to the log, because we dont need it at this point.
+                                */
+                                Log.info(packet.getFrom()+" "+element.getName()+"="+element.getStringValue());
```

Comment:
Unsure of the value of logging it.
If it's useful, should this be at debug, and include some context? Could be pretty noisy.

Reply:
Sorry was a debug output... forgot to delete it

Reply:
i will change it to debug output